pkg/cluster/acrtoken.go

```diff
@@ -108,6 +112,49 @@ func (m *manager) rotateACRTokenPassword(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
+	err = m.rotateOpenShiftConfigSecret(ctx, pullSecret.Data[corev1.DockerConfigJsonKey])
+	if err != nil {
+		return err
+	}
 
 	return nil
 }
+
+func (m *manager) rotateOpenShiftConfigSecret(ctx context.Context, encodedDockerConfigJson []byte) error {
+	openshiftConfigSecret, err := m.kubernetescli.CoreV1().Secrets(pullSecretName.Namespace).Get(ctx, pullSecretName.Name, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	updatedSecret := corev1.Secret{}
+
+	recreate := openshiftConfigSecret == nil ||
+		(openshiftConfigSecret.Type != corev1.SecretTypeDockerConfigJson || openshiftConfigSecret.Data == nil) ||
+		(openshiftConfigSecret.Immutable != nil && *openshiftConfigSecret.Immutable)
+
+	if recreate {
+		updatedSecret = corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      pullSecretName.Name,
+				Namespace: pullSecretName.Namespace,
+			},
+			Type: corev1.SecretTypeDockerConfigJson,
+			Data: make(map[string][]byte),
+		}
+	} else {
+		updatedSecret = *openshiftConfigSecret.DeepCopy()
+	}
+
+	updatedSecret.Data[corev1.DockerConfigJsonKey] = encodedDockerConfigJson
```

Comment:
This will overwrite all the data on the global pull secret, including customer data. We need to use [`pullsecret.Merge`](https://github.com/Azure/ARO-RP/blob/bfddd185cea94c279b14d87374d7933aeb25ecaa/pkg/util/pullsecret/pullsecret.go#L86)` at some point like the [controller code](https://github.com/Azure/ARO-RP/blob/bfddd185cea94c279b14d87374d7933aeb25ecaa/pkg/operator/controllers/pullsecret/pullsecret_controller.go#L173).

Reply:
we only need to merge if the original openshift-config pull secret had data in it, if we needed to recreate it, there is nothing to merge. I'll add the merge in for when we can update the secret.